fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalWindow.java

```diff
@@ -215,13 +218,34 @@ && child(0).child(0) instanceof LogicalPartitionTopN)) {
         long chosenPartitionLimit = Long.MAX_VALUE;
         long chosenRowNumberPartitionLimit = Long.MAX_VALUE;
         boolean hasRowNumber = false;
+        long atLeastLimit = -1;
         for (NamedExpression windowExpr : windowExpressions) {
             if (windowExpr == null || windowExpr.children().size() != 1
                     || !(windowExpr.child(0) instanceof WindowExpression)) {
                 continue;
             }
             WindowExpression windowFunc = (WindowExpression) windowExpr.child(0);
 
+            Optional<WindowFrame> windowFrame = windowFunc.getWindowFrame();
+            if (windowFrame.isPresent()) {
+                WindowFrame frame = windowFrame.get();
+                FrameUnitsType frameUnits = frame.getFrameUnits();
+                FrameBoundary rightBoundary = frame.getRightBoundary();
+                if (rightBoundary.getFrameBoundType() == FrameBoundType.UNBOUNDED_FOLLOWING) {
+                    return null;
+                } else if (frameUnits == FrameUnitsType.ROWS
```

Comment:
what about range frame?

Reply:
range frame only support `RANGE CURRENT ROW AND UNBOUNDED FOLLOWING` and `RANGE UNBOUNDED PRECEDING AND CURRENT ROW`